.github/workflows/basic.yml

```diff
@@ -145,6 +89,7 @@ jobs:
 
     - name: Configure build
       run: >
+        ${{ matrix.compiler.cxx == 'icpx' && '. /opt/intel/oneapi/setvars.sh &&' || ''}} 
```

Comment:
Could you set these vars in a oneAPI installation step instead? No additional check for icpx would be needed then.

Reply:
I'm not sure if variables set there would be propagated to any of the next steps...

Reply:
You are probably right if there is a new shell invoked for each step.